zebra/rtadv.c

```diff
@@ -632,45 +632,61 @@ static void rtadv_process_advert(uint8_t *msg, unsigned int len,
 
 	radvert = (struct nd_router_advert *)msg;
 
-	if ((radvert->nd_ra_curhoplimit && zif->rtadv.AdvCurHopLimit)
-	    && (radvert->nd_ra_curhoplimit != zif->rtadv.AdvCurHopLimit)) {
+#define SIXHOUR2USEC (int64_t)6 * 60 * 60 * 1000000
+
+	if ((radvert->nd_ra_curhoplimit && zif->rtadv.AdvCurHopLimit) &&
+	    (radvert->nd_ra_curhoplimit != zif->rtadv.AdvCurHopLimit) &&
+	    monotime_since(&zif->rtadv.lastadvcurhoplimit, NULL) >
```

Comment:
There's a bit of a problem here… `CLOCK_MONOTONIC` starts at 0 on boot, and you're initializing to 0.

⇒ the warning won't be printed in the first 6h after booting.

But those 6h are presumably when someone might be configuring this, and therefore need to be warned about the mismatch…

A `(!zif->rtadv.lastadvcurhoplimit.tv_sec || monotime_since(...) > SIXHOUR2USEC)` should do the trick.

Reply:
but it does, since it is initialized to 0, monotime_since will be greater than six hours and we will immediately see the first warning that happens on the box and then it will wait for 6 hours before warning

Reply:
> but it does, since it is initialized to 0, monotime_since will be greater than six hours

`monotime_since` won't be greater than six ours… the init value is 0, and the current value will be something like "5 minutes"…

Reply:
Why don't you try it?  It works properly

Reply:
```
CLOCK_MONOTONIC
              A nonsettable system-wide clock that represents monotonic
              time since—as described by POSIX—"some unspecified point
              in the past".  On Linux, that point corresponds to the
              number of seconds that the system has been running since
              it was booted.
```

Supposing we just booted 1 minute ago. CLOCK_MONOTONIC will be 60. `monotime_since(&zif->rtadv.lastadvcurhoplimit)` == `monotime_since(0)` will be 6e7, less than six hours.

Maybe it works properly on your system because your system happens to have a `CLOCK_MONOTONIC` that isn't set to 0 at boot, but regardless, @eqvinox's code is correct from a POSIX perspective and should be used.

Reply:
fixed